### ISSUE TITLE
API GridFieldAddNewButton should pass in a context for canCreate()

### ISSUE THREAD

**Issue by tractorcow** [silverstripe/silverstripe-framework#4913]
When adding a new element to a relation, sometimes the properties of that parent object would be necessary to check for create permissions on child elements. E.g. for SiteTree, canCreate takes an array `['Parent' => $parentPage]` as a second parameter.

It shouldn't be impossible to get the parent relation and ensure that $record is passed when checking permissions.

E.g.

``` php
public function getHTMLFragments($gridField) {
    $singleton = singleton($gridField->getModelClass());
    $list = $gridField->getList();
    $record = $list->methodThatGetsParentObject();
    $property = $list->methodThatGetsParentRelation();
    if(!$singleton->canCreate(null, array($property => $record))) {
        return array();
    }
    // Rest of code
}
```

Reference: https://github.com/silverstripe/silverstripe-userforms/pull/417

## Pull Requests

- [ ] https://github.com/silverstripe/silverstripe-framework/pull/6889

**Comment by hafriedlander** [comment]
Changing to alpha 2 as this doesn't seem specifically needed in alpha 1 (just before beta).

**Comment by flamerohr** [comment]
Franco and Simon Gow offered to work on this for hackday :)

**Comment by tractorcow** [comment]
Fixed with #6889
